dateutil/test/test_rrule.py

```diff
@@ -2669,6 +2670,19 @@ def testStr(self):
                           datetime(1998, 9, 2, 9, 0),
                           datetime(1999, 9, 2, 9, 0)])
 
+    def testStrWithTZID(self):
```

Comment:
I think you should add some tests for actually specifying the `tzinfos` parameter, and especially the case of `tzinfos={}` (which should fail to parse any `TZID` specified).